pdp/static/js/calendars.js

```diff
@@ -0,0 +1,818 @@
+/*globals classes */
+
+// TODO: This code is logical and works, but its API is hard to remember and
+// inconvenient to use. Refactor and/or add some convenience metods.
+
+(function () {
+    // Utility functions
+    
+    function toInt(string) {
+        if (!string) {
+            return undefined;
+        }
+        return parseInt(string, 10);
+    }
+
+    function isUndefined(v) {
+        return typeof v === 'undefined';
+    }
+
+    var isInteger = Number.isInteger;
+
+    var isArray = Array.isArray;
+
+    function isInRange(v, min, max) {
+        return min <= v && v < max;
+    }
+
+    function format(places, num) {
```

Comment:
Seems like you could have some weird results with decimals.
```javascript
> format(3, .967)
< "967"
```

Will this ever be used to format values from the time variable of a netCDF file? They are not always whole numbers.

```
netcdf pr_day_BCCAQv2+ANUSPLIN300_bcc-csm1-1-m_historical+rcp85_r1i1p1_19500101-21001231 {
dimensions:
    lon = 1068 ;
    lat = 510 ;
    time = 55115 ;
variables:
    double lon(lon) ;
    double lat(lat) ;
    double time(time) ;
        time:units = "days since 1850-01-01 00:00:00" ;
        time:calendar = "365_day" ;
        time:long_name = "Time" ;
        time:standard_name = "Time" ;
    short pr(time, lat, lon) ;
data:
 
 time = 36500.5, 36501.5, 36502.5, 36503.5, 36504.5, 36505.5, 36506.5,
...
}
```